parser/prism/Translator.cc

```diff
@@ -299,7 +299,19 @@ unique_ptr<parser::Node> Translator::translate(pm_node_t *node) {
                 statements = translateMulti(beginNode->statements->body);
             }
 
-            return make_unique<parser::Kwbegin>(location, move(statements));
+            if (statements.empty()) {
+                return make_node_with_expr<parser::Kwbegin>(MK::Nil(location), location, std::move(statements));
+            }
+
+            if (!hasExpr(statements)) {
```

Comment:
Do we need `!directlyDesugar` here as with the other cases?

```suggestion
            if (!directlyDesugar || !hasExpr(statements)) {
```